dbms/src/Core/Settings.h

```diff
@@ -102,6 +102,10 @@ struct Settings : public SettingsCollection<Settings>
     M(SettingBool, distributed_group_by_no_merge, false, "Do not merge aggregation states from different servers for distributed query processing - in case it is for certain that there are different keys on different shards.") \
     M(SettingBool, optimize_skip_unused_shards, false, "Assumes that data is distributed by sharding_key. Optimization to skip unused shards if SELECT query filters by sharding_key.") \
     \
+    M(SettingBool, enable_parallel_reading, true, "Enable parallel_reading for several data formats (JSON, TSV, TKSV, Values, CSV).") \
```

Comment:
Rename to `input_format_parallel_parsing`. Otherwise the name is extremely stupid.